File: src/com/sterndu/json/IntegerValue.java
package com.sterndu.json;

import java.util.function.Function;

public class IntegerValue implements NumberValue {

	private int value = 0;

	public IntegerValue(int value) {
		this.value = value;
	}

	@Override
	public Number getValue() { return value; }

	public int getValueInt() { return value; }

	public void setValue(int value) { this.value = value; }

	@Override
	public String toJson() {
		return value + "";
	}

	@Override
	public String toJson(Function<Object, String> function) {
		return value + "";
	}

	@Override
	public JsonValue toJsonValue() {
		return this;
	}

	@Override
	public JsonValue toJsonValue(Function<Object, String> function) {
		return this;
	}

	@Override
	public String toString() {
		return toJson();
	}

}

File: src/com/sterndu/json/IntegerValue.kt
@file:JvmName("IntegerValue")
package com.sterndu.json

class IntegerValue @JvmOverloads constructor(value: Int = 0) : NumberValue {
	private var valueInt = value

	fun getValueInt(): Int {
		return valueInt
	}

	override fun getValue(): Number {
		return valueInt
	}

	fun setValue(value: Int) {
		valueInt = value
	}

	override fun toJson(): String {
		return valueInt.toString() + ""
	}

	override fun toJson(function: (Any?) -> String): String {
		return valueInt.toString() + ""
	}

	override fun toJsonValue(): JsonValue {
		return this
	}

	override fun toJsonValue(function: (Any?) -> String): JsonValue {
		return this
	}

	override fun toString(): String {
		return toJson()
	}
}
